### PR TITLE
chore: Use eslint-plugin-jsdoc's recommended-typescript-error ruleset

### LINKED ISSUE #303: 🛠 Tooling: Use eslint-plugin-jsdoc's recommended-typescript-error ruleset
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

As of https://github.com/gajus/eslint-plugin-jsdoc/issues/997 -> https://github.com/gajus/eslint-plugin-jsdoc/pull/998, `eslint-plugin-jsdoc` now has a TypeScript preset rules config. It enables ESLint rules for good JSDoc practices that are still relevant in TypeScript code. Let's use it in this template!

### Additional Info

It'll need to be added in the ESLint config for `.ts` files: https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/4a4b8c5fb312698e98cedff07d2c0bb365f49b65/.eslintrc.cjs#L20

### PR DESCRIPTION
**Pull request by emday4prez**
- [x] Addresses an existing open issue: fixes #303 
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

ran ```pnpm install --save-dev eslint-plugin-jsdoc```, added ```"jsdoc"``` to eslint plugins array, and enabled the recommended rules in the eslint config *extends* array with ```"plugin:/jsdoc/recommended"```

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -7,6 +7,7 @@ module.exports = {
 	extends: [
 		"eslint:recommended",
 		"plugin:eslint-comments/recommended",
+
```
Comment (JoshuaKGoldberg) [inline .eslintrc.cjs:10]: [Nitpick] Unrelated change?

```suggestion
```

File: .eslintrc.cjs [hunk 2]
```diff
@@ -18,6 +19,7 @@ module.exports = {
 		},
 		{
 			extends: [
+				"plugin:jsdoc/recommended",
```
Comment (JoshuaKGoldberg) [inline .eslintrc.cjs:22]: Adding emphasis:

> chore: Use eslint-plugin-jsdoc's **recommended-typescript-error** ruleset

Close but not quite the right ruleset 😄. Search for `typescript` in https://github.com/gajus/eslint-plugin-jsdoc#configuration.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Gotta read these issues carefully 😉

File: .eslintrc.cjs [hunk 3]
```diff
@@ -18,6 +18,7 @@ module.exports = {
 		},
 		{
 			extends: [
+				"plugin:jsdoc/recommended-typescript",
```
Comment (JoshuaKGoldberg) [inline .eslintrc.cjs:21]: I can't comment on `src/greet.ts` because it's not part of the files changed - but this change introduced a complaint there. You can see it in the file viewer on GitHub and by running `pnpm lint` locally.

```plaintext
/home/runner/work/template-typescript-node-package/template-typescript-node-package/src/greet.ts
Warning:   5:8  warning  Missing JSDoc comment  jsdoc/require-jsdoc
```

I don't think we want to enforce that _every_ function has a JSDoc comment. So I think we should turn this rule off.

Comment (JoshuaKGoldberg) [inline .eslintrc.cjs:21]: > Adding emphasis:
> 
> > chore: Use eslint-plugin-jsdoc's **recommended-typescript-error** ruleset
> 
> Close but not quite the right ruleset 😄. Search for `typescript` in [gajus/eslint-plugin-jsdoc#configuration](https://github.com/gajus/eslint-plugin-jsdoc#configuration).

This is not the ruleset I asked for. 😞 

Do you see the difference?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
@emday4prez I really appreciate you sending these PRs, but requesting review when there are [checks failures](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks) shown by GitHub just means I'm going to ask you to fix them. 

I filed https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues/359 to make this more clear in the contributing docs

File: .eslintrc.cjs (hunk 3 above)
Comment (emday4prez) [reply .eslintrc.cjs]: sigh, yes. this has been updated.

Comment (emday4prez) [reply .eslintrc.cjs]: I added ```/* eslint-disable require-jsdoc */``` to ```src/greet.ts``` but it is still failing the same checks, so I have not yet found the solution. I am not exactly sure what the problem is, that is why I requested the review again earlier.

Comment (JoshuaKGoldberg) [reply .eslintrc.cjs]: > `/* eslint-disable require-doc */`

The rule's reporting under `jsdoc/require-jsdoc`, so you would need to disable that name. But I think we'd want to disable the rule in _all_ files, in case the user writes more code than just this one. You'll want to read up on https://eslint.org/docs/latest/use/configure/rules.

I also think we'll want all four of these `"off"`s:
https://github.com/JoshuaKGoldberg/ts-api-utils/blob/6050cfe20dcd07ac37b03d69c38801d1ce95ccb7/.eslintrc.cjs#L124-L127

> that is why I requested the review again earlier.

Makes sense! As a general piece of advice for PRs, I'd recommend being very explicit with what you're asking about. Post a comment in the PR with the question and include what you've tried along with what first-party (so here, on eslint.org) docs you've read.

**Comment by JoshuaKGoldberg** [comment]
👋 just checking in @emday4prez, is this still something you have time to work on? If so, anything I can help with?

**Comment by emday4prez** [comment]
> 👋 just checking in @emday4prez, is this still something you have time to work on? If so, anything I can help with?

Yes! I am away from my computer this weekend. I can finish this on Monday. Sorry about the delay! Thank you.

**Comment by JoshuaKGoldberg** [comment]
No rush, thanks! 🙌

File: src/greet.ts [hunk 4]
```diff
@@ -1,5 +1,9 @@
 import { GreetOptions } from "./types.js";
 
+/**
+ *
+ * @param options - The options for the greeting, or the message to greet.
+ */
```
Comment (JoshuaKGoldberg) [inline src/greet.ts:6]: Yeah these docs don't really add anything. And it's sample code anyway!

```suggestion
```

File: pnpm-lock.yaml [hunk 5]
```diff
@@ -15,7 +15,7 @@ devDependencies:
     version: 5.48.2(eslint@8.32.0)(typescript@5.0.2)
   '@vitest/coverage-istanbul':
     specifier: ^0.30.0
-    version: 0.30.1(vitest@0.30.0)
+    version: 0.30.0(vitest@0.30.0)
```
Comment (JoshuaKGoldberg) [inline pnpm-lock.yaml:18]: Interesting how a lot of these versions have _decreased_ in this lockfile. I can take a look.

**Review by JoshuaKGoldberg — APPROVED** [review]
🚀 super, thanks @emday4prez! Really excited to have this in 😄

File: .eslintrc.cjs [hunk 6]
```diff
@@ -100,6 +102,10 @@ module.exports = {
 		"simple-import-sort/imports": "error",
 
 		// These on-by-default rules don't work well for this repo and we like them off.
+		"jsdoc/require-jsdoc": "off",
+		"jsdoc/require-param": "off",
+		"jsdoc/require-property": "off",
+		"jsdoc/require-returns": "off",
```
Comment (JoshuaKGoldberg) [inline .eslintrc.cjs:108]: The `jsdoc` plugin was declared in an `overrides` for `**/*.ts` files, so it overrides any root-level rule configurations. I'll move these to the overrides.

fyi @emday4prez

File: pnpm-lock.yaml (hunk 5 above)
Comment (JoshuaKGoldberg) [reply pnpm-lock.yaml]: cc @emday4prez - in case you ever see this odd change again (where a branch's lockfile has a bunch of package versions behind the main branch), my steps to resolve it are:

1. Reset the lockfile back to the `main` branch
2. Re-run installation

```shell
git checkout main -- pnpm-lock.yaml
pnpm install
```

File: .eslintrc.cjs (hunk 6 above)
Comment (JoshuaKGoldberg) [reply .eslintrc.cjs]: f593225
